### PR TITLE
Implementation Of Conference Information View

### PR DESCRIPTION
**Pull request by at-robins**
A new view displaying conference information has been defined in the template info.scala.html.
The conference information stored on the server is Markdown formatted therefore the commonmark library is used in the server side conference model to convert the data to HTML.
All tests perform without error, however a encoding conflict inside the database seems to cause erroneous display of some special characters such as ä, ü, ö, etc..

### CODE REVIEW

**Comment by cgars** [comment]
## Some general comments
* Well done!
* Rebase again, and push (maybe with force) this PR instead of creating a new one.
* Only show the Information stuff if there is data.
* Maybe we could have the info view not as an additional tab but below the conference (extended) view

File: app/controllers/Application.scala [hunk 1]
```diff
@@ -43,6 +43,17 @@ class Application(implicit val env: Environment[Login, CachedCookieAuthenticator
     Ok(views.html.submission(user, conf, None))
   }
 
+  /**
+    * Handles the display of the information tab for a specific conference.
+    *
+    * @param confId - the conference ID
```
Comment (cgars) [inline app/controllers/Application.scala]: For the sake of uniformity, I would probably go without doc here.

File: app/models/Conference.scala [hunk 2]
```diff
@@ -123,6 +130,21 @@ class Conference extends Model with Owned with Tagged {
   def touch (): Unit = {
     this.mtime = new DateTime(DateTimeZone.UTC)
   }
+
+  /**
```
Comment (cgars) [inline app/models/Conference.scala]: same here

File: app/models/Conference.scala [hunk 3]
```diff
@@ -123,6 +130,21 @@ class Conference extends Model with Owned with Tagged {
...
+    *
+    * @return the HTML representation of the conference information
+    */
+  def getInfoAsHTML () : String = {
+    /*
+     * Create a HTML renderer and a Markdown parser using factory methods.
+     * Parse the Markdown formatted conference information and render it to HTML.
+     */
+    // TODO: write test case
+    return HtmlRenderer.builder().build().render(Parser.builder().build().parse(this.info))
```
Comment (cgars) [inline app/models/Conference.scala]: We could consider not instantiating this all the time. Good enough for the time being though.

**Review by cgars — CHANGES_REQUESTED** [review]

**Comment by at-robins** [comment]
The conference information view is currently implemented as second box below the conference box. Maybe I should change this to be the same as the "Administration" function when logged in? I guess it comes down to personal taste.

**Review by cgars — APPROVED** [review]

**Review by cgars — CHANGES_REQUESTED** [review]
Way better now. 
I have an issue with the reuse of the info view. We have the conference uuid in there twice now. Also, the "no info here" warning is kind of obsolete

**Comment by at-robins** [comment]
I removed the view completely now.
Should I write a test case for the Markdown-parsing function? It is pretty much just a few library calls, so I do not think it to be really necessary.

File: app/views/conference.scala.html [hunk 4]
```diff
@@ -77,4 +77,16 @@ <h3>Administration</h3>
             </div>
         }
     </div>
+
+    <!-- The conference info panel -->
+    @if(conference.info != null) {
```
Comment (cgars) [inline app/views/conference.scala.html]: Have you actually checked whether this is working? I kind of doubt that it will always be null if empty. You could start by testing for such cases

File: test/models/ConferenceTest.scala [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+    }
+  }
+
+  @Test
+  def testGetInfoAsHTML() : Unit = {
+    /*
+     * Mostly a placeholder for potential future changes, which need to be tested differently.
+     */
+    val r : Random = new Random()
+    var testingInfo : String = ""
```
Comment (cgars) [inline test/models/ConferenceTest.scala]: Testing this (empty string) would actually be an interesting corner case

File: test/models/ConferenceTest.scala [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+    var testingConf : Conference = null
+    for (i <- 0 to 10000) {
+      testingInfo = ConferenceTest.generateRandomString(r.nextInt(200))
+      testingConf = Conference(Some("uuid"), Some("someconf"), Some("XX"), Some("G"), Some("X"),
+        None, None, Some(false), Some(true), Some(false), Some(true), None, None, None, info = Some(testingInfo))
+      assert(testingConf.getInfoAsHTML() == Conference.convertMarkdownToHTML(testingInfo))
+    }
+    // test nulls throwing a NullPointerException
```
Comment (cgars) [inline test/models/ConferenceTest.scala]: But should they. I know they do however should they really.  Are you not, you trying to avoid it?

File: test/models/ConferenceTest.scala (hunk 5 above)
Comment (at-robins) [reply test/models/ConferenceTest.scala]: Actually empty strings are tested implicitly at two points. First in 
```
  def testConvertMarkdownToHTML() : Unit = {
    val r : Random = new Random()
    // test nulls throwing a NullPointerException
    intercept[NullPointerException] {
      Conference.convertMarkdownToHTML(null)
    }
    // test empty markdown returning empty html
    assert(Conference.convertMarkdownToHTML("") == "")
```
, which is currently equivalent to the instance functions.
And a second time as part of the random testing since strings of length zero are allowed.
I, however, added the empty string corner case explicitly to the instance function tests for the sake of completeness.

File: test/models/ConferenceTest.scala (hunk 6 above)
Comment (at-robins) [reply test/models/ConferenceTest.scala]: Sure, I guess you mean the XSS protection is still missing?

Comment (cgars) [reply test/models/ConferenceTest.scala]: Let me ask differently. where would the np happen and how could it be avoided?

**Review by cgars — APPROVED** [review]
